observability/micrometer/src/main/java/org/wildfly/extension/micrometer/MicrometerResourceDefinitionRegistrar.java

```diff
@@ -0,0 +1,136 @@
+    public static final String COLLECTOR_NAME = MICROMETER_MODULE + ".wildfly-collector";
+    public static final String WILDFLY_REGISTRY_NAME = MICROMETER_MODULE + ".registry";
+
+    public static final RuntimeCapability<Void> MICROMETER_COLLECTOR_RUNTIME_CAPABILITY =
+            RuntimeCapability.Builder.of(COLLECTOR_NAME, MicrometerCollector.class)
+                    .addRequirements(CLIENT_FACTORY_CAPABILITY, MANAGEMENT_EXECUTOR, PROCESS_STATE_NOTIFIER)
+                    .build();
+    public static final RuntimeCapability<Void> MICROMETER_REGISTRY_RUNTIME_CAPABILITY =
+            RuntimeCapability.Builder.of(WILDFLY_REGISTRY_NAME, WildFlyRegistry.class)
+                    .build();
+
+    @Deprecated
+    static final SimpleAttributeDefinition ENDPOINT = SimpleAttributeDefinitionBuilder
+            .create(MicrometerConfigurationConstants.ENDPOINT, ModelType.STRING)
+            .setAttributeGroup(MicrometerConfigurationConstants.OTLP_REGISTRY)
+            .setRequired(false)
+            .setAllowExpression(true)
+            .setRestartAllServices()
+            .build();
+
+    @Deprecated
+    static final SimpleAttributeDefinition STEP = SimpleAttributeDefinitionBuilder
+            .create(MicrometerConfigurationConstants.STEP, ModelType.LONG, true)
+            .setAttributeGroup(MicrometerConfigurationConstants.OTLP_REGISTRY)
+            .setDefaultValue(new ModelNode(TimeUnit.MINUTES.toSeconds(1)))
+            .setMeasurementUnit(MeasurementUnit.SECONDS)
+            .setAllowExpression(true)
+            .setRestartAllServices()
+            .build();
+
+    static final StringListAttributeDefinition EXPOSED_SUBSYSTEMS =
+            new StringListAttributeDefinition.Builder("exposed-subsystems")
+                    .setDefaultValue(ModelNode.fromJSONString("[\"*\"]"))
+                    .setRequired(false)
+                    .setRestartAllServices()
+                    .build();
+
+    static final AttributeDefinition[] ATTRIBUTES = {
+            ENDPOINT, STEP, EXPOSED_SUBSYSTEMS
+    };
+    private WildFlyCompositeRegistry wildFlyRegistry;
+    private final ResourceRegistration registration;
+    private final ResourceDescriptor descriptor;
+    static final ParentResourceDescriptionResolver RESOLVER =
+            new SubsystemResourceDescriptionResolver(MicrometerConfigurationConstants.NAME,
+                    MicrometerResourceDefinitionRegistrar.class);
+
+    MicrometerResourceDefinitionRegistrar() {
+        ResourceDescriptor.Builder builder =
+                ResourceDescriptor.builder(RESOLVER.createChildResolver(MicrometerSubsystemRegistrar.PATH));
+
+        registration = ResourceRegistration.of(MicrometerSubsystemRegistrar.PATH);
+        descriptor = builder.addCapability(MICROMETER_COLLECTOR_RUNTIME_CAPABILITY)
+                .addCapability(MICROMETER_REGISTRY_RUNTIME_CAPABILITY)
+                .build();
```

Comment:
I would move this to your register(...) method.  These don't need to be member variables.